src/api/app/assets/javascripts/webui2/staging_workflow.js

```diff
@@ -0,0 +1,11 @@
+function setSpinnersForDeletion() { // jshint ignore:line
+  $(".delete-spinner").hide();
+
+  $(document).ajaxStart(function() {
+    $(".delete-spinner").show();
```

Comment:
This only works as long as there is only one element with this class. Also this event handler would be triggered with any ajax call that is send to the document.

I would try to attach the event handler to something more specific instead, eg. the modal or the form.
One way to do this would be this:
```javascript
$(document).on('submit', '#staging-delete', function(event) { 
  $(this).find('.delete-spinner').removeClass('d-none');
}
```

But there might be better solutions.

Reply:
@krauselukas Give it a try with [automatic-disabling](https://guides.rubyonrails.org/working_with_javascript_in_rails.html#automatic-disabling). This looks like what we want here.

The [unobtrusive javascript](https://guides.rubyonrails.org/working_with_javascript_in_rails.html#unobtrusive-javascript) section might be a good read as well :smile:

Reply:
> This only works as long as there is only one element with this class.

How so? .show will show all elements matched